operator/src/test/java/org/keycloak/operator/testsuite/unit/PodTemplateTest.java

```diff
@@ -92,9 +83,10 @@ public Map<String, String> podLabels() {
 
         kc.setSpec(keycloakSpecBuilder.build());
 
-        var deployment = new KeycloakDeployment(null, config, kc, existingDeployment, "dummy-admin");
+        Context<Keycloak> context = Mockito.mock(Context.class);
+        Mockito.when(context.getSecondaryResource(StatefulSet.class)).thenReturn(Optional.ofNullable(existingDeployment));
```

Comment:
Keycloak tries to stay away from Mockito. Even though it might make sense to use it here, we'd be creating a precedence for the project. :) Could we perhaps try keeping the previous approach, i.e. implement a custom class implementing the Context interface?

Reply:
> implement a custom class implementing the Context interface?

Isn't that just boiler plate for the sake of saying that mockito won't be used?  Wouldn't it be better to just enforce some principled use of Mockito?